### PR TITLE
OCPBUGS-21409: upgrade golang.org/x/net to 0.17.0 to address CVE-2023…

### PR DESCRIPTION
**Pull request by machine424**
…-39325

### CODE REVIEW

**Comment by machine424** [comment]
Seems to be an incompatibility between some new `golang.org/x/sys` modules needed by `golang.org/x/net v0.17.0` and go1.16 used for 4.10.

**Comment by machine424** [comment]
Upgraded the go versions in go.mod to to the one used for building to avoid some incompatibility issues [https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_k8s-prometh[…]rometheus-adapter-release-4.11-e2e-aws/1713861027369586688](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_k8s-prometheus-adapter/84/pull-ci-openshift-k8s-prometheus-adapter-release-4.11-e2e-aws/1713861027369586688)

File: go.mod [hunk 1]
```diff
@@ -21,3 +21,88 @@ require (
 	sigs.k8s.io/custom-metrics-apiserver v1.22.0
 	sigs.k8s.io/metrics-server v0.5.0
 )
+
+require (
+	github.com/NYTimes/gziphandler v1.1.1 // indirect
```
Comment (machine424) [inline go.mod:26]: this, e,g, was removed fro the vendor/, don't know if it's normal yet

Comment (machine424) [reply go.mod:26]: Seems normal it's the case also on the 4.12 branch.

**Comment by simonpasquier** [comment]
/hold

holding for now

**Pull request closed without merging**